src/inmanta/data/__init__.py

```diff
@@ -1634,6 +1654,48 @@ async def insert_many(
         async with cls.get_connection(connection) as con:
             await con.copy_records_to_table(table_name=cls.table_name(), columns=columns, records=records, schema_name="public")
 
+    @classmethod
+    async def insert_many_with_overwrite(
+        cls, documents: Sequence["BaseDocument"], *, connection: Optional[asyncpg.connection.Connection] = None
+    ) -> None:
+        """
+        Insert new documents. If the document already exists, overwrite it.
+        """
+        if not documents:
+            return
+        column_names = cls.get_field_names()
+        primary_key_fields = cls._get_names_of_primary_key_fields()
+        primary_key_string = ",".join(primary_key_fields)
+        update_set = list(set(column_names) - set(cls._get_names_of_primary_key_fields()))
+        update_set_string = ",\n".join([f"{item} = EXCLUDED.{item}" for item in update_set])
+
+        values = []
+        for record in documents:
+            col_names, dao_values = record._get_column_names_and_values()
+            values.append(dao_values)
+
+        column_names_as_sql_string = ", ".join(column_names)
+
+        start_value = 1
+        placeholders = []
+        for _ in range(len(values)):
+            sublist = []
+            step = len(values[0])
+            for i in range(start_value, start_value + step):
+                sublist.append(f"${i}")
+            placeholders.append("(" + ", ".join(sublist) + ")")
+            start_value += step
+        values_placeholder = ", ".join(placeholders)
```

Comment:
This could maybe use a comment as to what it is doing ? And/or I find this more readable with list comprehension: (not a strong opinion at all, but the comment would still be nice)
```suggestion
n_docs = ...
n_cols = ...
placeholders = ', '.join(
    [
        '(' + ', '.join([f'${doc*n_cols+col}' for col in range(1, n_cols+1)]) + ')' for doc in range(n_docs)
    ]
)

```

Reply:
Do you still find a comment is required? I find the variable name clear enough and I don't know what to put in the comment

Reply:
No, it's fine without a comment with the list comprehension syntax :+1: